src/headless/converse-muc.js

```diff
@@ -2436,6 +2436,12 @@ converse.plugins.add('converse-muc', {
                     'num_unread': 0,
                     'num_unread_general': 0
                 });
+            },
+
+            async onBanUser (message, reason) {
+                const jid = message.get('from');
+                const room = await api.rooms.get(''); //@TODO get proper room
+                room.sendAffiliationIQ('outcast', { jid, reason });
```

Comment:
You created the `onBanUser` method on the ChatRoom model, so there's no need to get the room via the api, you can just call `this.sendAffiliationIQ('outcast', { jid, reason })`.